extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/ParquetTableLocation.java

```diff
@@ -667,19 +672,33 @@ private PushdownResult pushdownMatchFilter(List<Integer> parquetIndices, Pushdow
      * Apply the filter to the data index table and return the result.
      */
     @NotNull
-    private PushdownResult pushdownDataIndex(WhereFilter filter, BasicDataIndex dataIndex, PushdownResult result) {
+    private PushdownResult pushdownDataIndex(
+            final WhereFilter filter,
+            final Map<String, String> renameMap,
+            final BasicDataIndex dataIndex,
+            final PushdownResult result) {
         final RowSetBuilderRandom matchingBuilder = RowSetFactory.builderRandom();
-        try (final SafeCloseable scope = LivenessScopeStack.open()) {
-            // Data type changes handled by the filter copy/init
+        try (final SafeCloseable ignored = LivenessScopeStack.open()) {
             final WhereFilter copiedFilter = filter.copy();
             copiedFilter.init(dataIndex.table().getDefinition());
 
-            // TODO: handle renames
-            final Table filteredTable = dataIndex.table().where(copiedFilter);
+            final Collection<io.deephaven.api.Pair> renamePairs = renameMap.entrySet().stream()
```

Comment:
Does pair have a type parameter we should use?

Reply:
No, this is not a generic but a Pair of two `ColumnName`s